src/util/tekton-vfs.ts

```diff
@@ -0,0 +1,159 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+import { FileSystemProvider, EventEmitter, FileChangeEvent, Event, Uri, Disposable, FileStat, FileType, FileChangeType } from 'vscode';
+import { CliExitData, cli } from '../cli';
+import { newK8sCommand, getStderrString } from '../tkn';
+import * as path from 'path';
+import * as os from 'os'
+import * as fsx from 'fs-extra';
+import { VirtualDocument } from '../yaml-support/yaml-locator';
+import { TektonItem } from '../tekton/tektonitem';
+
+export const TKN_RESOURCE_SCHEME = 'tekton';
+export const TKN_RESOURCE_SCHEME_READONLY = 'tekton-ro';
+const readonlyRegex = /(taskrun|pipelinerun)/ as RegExp;
+
+/**
+ * Create Uri for Tekton VFS 
+ * @param type tekton resource type
+ * @param name tekton resource name
+ * @param format output format (yaml|json)
+ */
+export function tektonFSUri(type: string, name: string, format: string): Uri {
+  const scheme = readonlyRegex.test(type) ? TKN_RESOURCE_SCHEME_READONLY : TKN_RESOURCE_SCHEME;
+  return Uri.parse(`${scheme}://kubernetos/${type}/${name}.${format}`);
```

Comment:
`kubernetos ` -> something related to `tekton`

Reply:
No, as it is kubernetos resources, and we use `kubectl` to fetch it, we relay to tekton in sheme